### DISCUSSION TITLE
Next major version to v0.9 and versioning

### DISCUSSION THREAD

**Discussion by metalefty** [neutrinolabs/xrdp#2274] [Ideas]
As we're making big breaking changes such as authentication refactor or GFX into devel branch, I already decided not to merge such changes into v0.9 versions. Such big changes are shipped to next major version. Let's decide what the next version is.

There are two main candidates:
- v0.10 (start from v0.10.0)
- v1.0 (start from v1.0.0)

I personally would like to choose v1.0 and bump second-level version for each regular releases (about 3 times per year). I was inspired this versioning by [TigerVNC](https://github.com/TigerVNC/tigervnc/releases).  This looks like [Semantic Versioning](https://semver.org/) but I don't intend to introduce strict Semantic Versioning.  Because Semantic Versioning requires to declare a public API.  Doing it with our limited resources is a dauting work.

- v1.0.0
    - v1.0.1 (minor bugfixes to v1.0.0)
    - v1.0.80 (dev version to be next v1.1.0)
- v1.1.0

One concern to choose v1.0 is it implies stable, better quality than v0.9. Actually, v1.0 takes a few step forward to v0.9 and has many advanced features so v0.9 is more stable. We need to note this point on the release note.

**Comment by matt335672** [comment]
I don't have a strong preference either way, so I'd be happy to go with v1.0.0

Maybe something like this on the release note?

-----
## Versioning number change
This is a major release of xrdp with new features, and a few incompatibilities (see below)

We are taking this opportunity to bring our version numbering in-line with projects such as TigerVNC. From this version, the version string vx.y.z means the following:-

| field | meaning |
|:----- |:------- |
|x| Major release. Increased for changes which are not backwards compatible |
|y| Minor release. Increased for changes which are backwards compatible. Most regular releases will be in this category |
|z (< 80) | Point release. Compatible security and bug fixes |
|z (>= 80) | For development only. Do not use in production. |

Despite appearances, this is not [semantic versioning](https://semver.org/).

-------

That's only a first draft.

**Reply by metalefty** [reply]
Thanks. That is a good draft.

**Comment by metalefty** [comment]
I 'm thinking of introducing v0.9 branch in xorgxrdp to keep compatibility. See also neutrinolabs/xorgxrdp#212.

For simplicity, the next release of xorgxrdp v0.2.z will be v0.9.20. This doesn't mean versions of xrdp and xorgxrdp are in strict sync. `z` might be different but it simply indicates xorgxrdp v0.9 is for xrdp v0.9, xorgxrdp v1 is for xrdp v1.
